apps/dot_ext/loggers.py

```diff
@@ -0,0 +1,199 @@
+                application = None
+
+        if application:
+            # Create and return AuthFlowUuid instance for tracking.
+            AuthFlowUuid.objects.create(auth_uuid=new_auth_uuid,
+                                        client_id=application.client_id,
+                                        auth_pkce_method=auth_pkce_method)
+
+
+def get_session_auth_flow_trace(request):
+    '''
+    Function to get auth flow related items from the
+    session.
+
+    Returns a auth_flow_dict type DICT of values for logging.
+    '''
+    if request:
+        auth_flow_dict = {}
+        for k in SESSION_AUTH_FLOW_TRACE_KEYS:
+            auth_flow_dict[k] = request.session.get(k, None)
+        return auth_flow_dict
+    else:
+        return {}
+
+
+def set_session_auth_flow_trace(request, auth_flow_dict):
+    '''
+    Function to set auth flow related items in the
+    session from a dictionary.
+    '''
+    if request:
+        for k in SESSION_AUTH_FLOW_TRACE_KEYS:
+            request.session[k] = auth_flow_dict.get(k, None)
+
+
+def set_session_values_from_auth_flow_uuid(request, auth_flow_uuid):
+    '''
+    Function to set auth flow related items in the
+    session given an AuthFlowUuid instance.
+    '''
+    application = None
+    if auth_flow_uuid:
+        request.session['auth_uuid'] = str(auth_flow_uuid.auth_uuid)
+        request.session['auth_pkce_method'] = str(auth_flow_uuid.auth_pkce_method)
+
+        # Get the application.
+        Application = get_application_model()
+        try:
+            application = Application.objects.get(client_id=auth_flow_uuid.client_id)
+
+            # Set values in session.
+            request.session['auth_app_id'] = str(application.id)
+            request.session['auth_app_name'] = str(application.name)
+            request.session['auth_client_id'] = str(application.client_id)
+        except Application.DoesNotExist:
+            pass
+
+
+def update_session_auth_flow_trace(request=None, auth_uuid=None, state=None, code=None):
+    '''
+    Function for updating auth flow related items in the
+    AuthFlowUuid tracking object and session.
+
+    Summary of cases based on arguments passed in:
+
+    - auth_uuid and code = Update previously created AuthFlowUuid instance with code.
+    - code =  Get session values from AuthFlowUuid via code.
+    - auth_uuid = Get session values from AuthFlowUuid.
+    - state = Update or set state in session.
+    '''
+    # Set to None for Unit tests to work.
+    auth_flow_uuid = None
+
+    if auth_uuid and code:
+        # Get and update previously created AuthFlowUuid instance with code.
+        try:
+            auth_flow_uuid = AuthFlowUuid.objects.get(auth_uuid=auth_uuid)
+            # Set code.
+            auth_flow_uuid.code = code
+            auth_flow_uuid.save()
+        except AuthFlowUuid.DoesNotExist:
+            # Create AuthFlowUuid instance, if it doesn't exist.
+            if auth_uuid:
```

Comment:
If I'm not mistaken this is a redundant if since its already verified to be not None in the first if statement on line 143.